salt/grains/lvm.py

```diff
@@ -1,5 +1,13 @@
 """
 Detect LVM Volumes
+
+.. versionchanged:: 3006
+
+To disable these grains set `enable_lvm_grains: False` in the minion config.
+
+.. code_block:: yaml
+
+    enable_lvm_grains: False
```

Comment:
We could use a global config instead, that would make things a bit easier for the user. There already is the `grains_blacklist` which is similar in concept but works on a different granularity. [`grains_blacklist`](https://github.com/saltstack/salt/blob/474ea03bb49c9aa161fefc797f994526e02fb3ca/salt/loader/__init__.py#L1166) only filters the grains *after* they are computed, which does not help when grains computation is slow. But it allows, afaics, to filter out single grains out of a grain function that computes multiple grains. If every grain function only ever returned a single grain, we could just skip the computation for those functions based on the blacklist.

What do you think about the following:
1. we add `disable_grains` 
2. we add a "config translation layer" that automatically populates `disable_grains` from all existing "grains opt-out settings"
3. all grain modules that use these settings are changed to only check the single `disable_grains` list
4. we mark existing "grains opt-out settings" deprecated, telling users to use the `disable_grains` list instead

In an unrelated project, I just experienced how changing the configuration can go badly when old config keys are not accepted anymore and I'd keep 2. around for a long time to not break existing configs.

*Depending on how we implement 2., `enable_lvm_grains: False` could still work without explicitly allowing it. That it's of little interest to you, but could be useful in our downstream Salt package*

Reply:
Can you elaborate on (we add a "config translation layer" that automatically populates disable_grains from all existing "grains opt-out settings")

I just want to be clear what your proposing on this one.

There other points sound solid to me.

Reply:
Sure, I have two ways to do this in mind, either using a regex to extract the grains module from the existing "grains opt-out settings", or with a hard-coded list of known "opt-out settings". The overall approach is the same for both, if it's possible with a simple regex I'd prefer that approach since it's more open. Anyway, the idea is this:

At the end of `apply_minion_config` in `salt/config/__init__.py`, the existing config keys are checked (using the list or regex). For each match that's set to `False`, the grain module name is added to `disable_grains`.

Reply:
I started implementing this and hit a bit of a blocker: some grains are currently disabled by default. I'm not sure how to handle this. If I add them to `disabled_grains` per default (in `DEFAULT_(MASTER|MINION)_OPTS`, the user can't take them out using a config file.

One way to solve this is to only add the default `disabled_grains` in `conf/master` and `conf/minion`, but that's not a nice solution. Another way would be to also add an `enabled_grains` (or similarly named) config, with the sole purpose of taking grains out of `disabled_grains`.

How do you feel about this @Ch3LL ?

Reply:
Please disregard my previous comment, I didn't see the forest for the trees, the user can simply provide `disabled_grains` with the default(s) removed.